R/output_combined_table.R

```diff
@@ -0,0 +1,137 @@
+#' Create combined traits.build table
+#'
+#' Create a single database output that merges together the information
+#' in all relational tables within a traits.build database.
+#' Trait measurements are still output in long format (1 row per trait value),
+#' but all measurement-related metadata (methods, location properties, context properties, contributors)
+#' are now included as additional columns in a single table.
+#'
+#' @param database A traits.build database
+#'
+#' @return
+#' @export
+#'
+#' @examples
+database_create_combined_table <- function(database) {
+
+  location_latlon <-
+    database$locations %>%
+      dplyr::filter(location_property %in% c("latitude (deg)", "longitude (deg)")) %>%
+      tidyr::pivot_wider(names_from = location_property, values_from = value)
+
+  location_properties <-
+    database$locations %>%
+      dplyr::filter(!location_property %in% c("latitude (deg)", "longitude (deg)")) %>%
+      dplyr::mutate(
+        location_property = stringr::str_replace_all(location_property, "=", "-"),
+        value = stringr::str_replace_all(value, "=", "-"),
+        location_property = stringr::str_replace_all(location_property, ";", ","),
+        value = stringr::str_replace_all(value, ";", ",")
+      ) %>%
+      dplyr::mutate(location_properties = paste0(location_property, "=", value)) %>%
+      dplyr::select(dplyr::all_of(c("dataset_id", "location_id", "location_name", "location_properties"))) %>%
+      dplyr::group_by(dataset_id, location_id, location_name) %>%
+      dplyr::mutate(location_properties = paste0(location_properties, collapse = "; ")) %>%
```

Comment:
possibly but, I think `distinct()` is actually clearer in this case - otherwise you'd just have to list all the column individually and decide how to treat each one (lots of "take first value")